File: problem_five/5.js
(function doMagic (max) {
  var number = max;
  while(true){
    if (divisible_by_range(number, max)) {
      console.log(number);
      return number;
    }
    number ++;
  }
})(20);

function divisible_by_range (number, max) {
  for (var i=1; i<max; i++){
    if (number % i !==0) {return false;}
  }
  return true;
}

File: problem_five/5.rb
def do_magic(range)
  number = range.last
  loop do |_|
    return puts(number) if divisible_by_range?(number, range)
    number += 1
  end
end

def divisible_by_range?(number,range)
  range.all? { |divisor| number % divisor == 0 }
end

do_magic([*1..20])